week1/logic-brute/borboss366/Main.hs

```diff
@@ -0,0 +1,104 @@
+module Main where
+
+newtype SRCRange = SRCRange {getRange :: (Int, Int)} deriving (Eq, Show)
+newtype PairValue = PairValue {getPair :: (Int, Int)} deriving (Eq, Show)
+
+a :: PairValue -> Int
+a = fst.getPair
+
+b :: PairValue -> Int
+b = snd.getPair
+
+prdVal :: PairValue -> Int
+prdVal pv = a pv * b pv
+
+smVal :: PairValue -> Int
+smVal pv = a pv + b pv
+
+rs :: SRCRange -> Int
```

Comment:
You could make `SRCRange` a record and then `rs` with `re` would've been record fields.